### PR TITLE
feat(SpEL): implement to configure the limit of characters for SpEL expressions

### PR DESCRIPTION
**Pull request by j-sandy**
Spring Expression Lanuage (SpEL) has a default limit of 10,000 characters. Springframework provides the feature to configure the limit. This feature allows to configure the limit of characters for SpEL expressions.

Approach:
In order to use an expression with characters more than the given default limit, require to follow either of the below approaches:
1. For Springframework >=5.3.28 and <6.1.3, by setting `maximumExpressionLength` field while instantiating the custom `SpelParserConfiguration` class. 
https://github.com/spring-projects/spring-framework/issues/30380 
https://github.com/spring-projects/spring-framework/issues/30446

3. For Springframework >=6.1.3, by setting a JVM system property or Spring property named `spring.context.expression.maxLength` to the maximum expression length needed by your application. 
https://github.com/spring-projects/spring-framework/issues/31952 
https://github.com/spring-projects/spring-framework/commit/785598629abda944343a02307ad82a79bb31b589

Spinnaker supports spring boot 2.7.18, that brings springframework 5.3.31 [https://docs.spring.io/spring-boot/docs/2.7.18/reference/html/dependency-versions.html#appendix.dependency-versions.propertie9]. So first approach need to be implemented along with spinnaker enhancement to expose the `maximumExpressionLength` field.

### CODE REVIEW

File: echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java [hunk 1]
```diff
@@ -26,11 +29,22 @@
...
   private final ObjectMapper mapper;
-  private final ExpressionParser parser = new SpelExpressionParser();
+  private final ExpressionParser parser;
   private final ParserContext parserContext = new TemplateParserContext("${", "}");
 
-  public ExpectedArtifactExpressionEvaluationPostProcessor(ObjectMapper mapper) {
+  public ExpectedArtifactExpressionEvaluationPostProcessor(
+      ObjectMapper mapper, @Autowired ExpressionProperties expressionProperties) {
     this.mapper = mapper;
+    if (expressionProperties != null) {
```
Comment (dbyron-sf) [inline echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java:38]: Without specifying required = false, I don't see how expressionProperies can be null.

File: echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java [hunk 2]
```diff
@@ -26,11 +29,22 @@
 @Component
 public class ExpectedArtifactExpressionEvaluationPostProcessor implements PipelinePostProcessor {
   private final ObjectMapper mapper;
-  private final ExpressionParser parser = new SpelExpressionParser();
+  private final ExpressionParser parser;
   private final ParserContext parserContext = new TemplateParserContext("${", "}");
 
-  public ExpectedArtifactExpressionEvaluationPostProcessor(ObjectMapper mapper) {
+  public ExpectedArtifactExpressionEvaluationPostProcessor(
+      ObjectMapper mapper, @Autowired ExpressionProperties expressionProperties) {
```
Comment (dbyron-sf) [inline echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java:36]: Is `@Autowired` required here.  Doesn't everything get autowired via `@Component`?

File: echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java (hunk 1 above)
Comment (j-sandy) [reply echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java]: Updated the code.

File: echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java (hunk 2 above)
Comment (j-sandy) [reply echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java:36]: Yes, it is required as we are using it in constructor of a component and concluding its use inside the constructor.

Comment (dbyron-sf) [reply echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java:36]: But...we don't have it for ObjectMapper.  Why is ExpressionProperties different?

Comment (j-sandy) [reply echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/postprocessors/ExpectedArtifactExpressionEvaluationPostProcessor.java:36]: I verified without using `@Autowired`, build and tests are passing. So, updating code as suggested.

**Comment by dbyron-sf** [comment]
@mergifyio update

**Review by dbyron-sf — APPROVED** [review]

**Pull request merged**